app/controllers/sidebar_sections_controller.rb

```diff
@@ -52,4 +53,12 @@ def check_if_member_of_group
       raise Discourse::InvalidAccess
     end
   end
+
+  private
+
+  def check_access_if_public
+    if params[:public] && !@guardian.can_create_public_sidebar_section?
```

Comment:
```suggestion
    guardian.ensure_can_create_public_sidebar_section! if params[:public]
```

You don't need `@` for `guardian` in controllers either